src/frontends/pytorch/src/op/linear.cpp

```diff
@@ -15,9 +16,23 @@ OutputVector translate_linear(const NodeContext& context) {
     // schema: aten::linear(Tensor input, Tensor weight, Tensor? bias=None) -> Tensor
     num_inputs_check(context, 2, 3);
     auto x = context.get_input(0);
-    auto y = context.get_input(1);
-    auto matmul = context.mark_node(std::make_shared<ov::op::v0::MatMul>(x, y, false, true));
-    return {context.mark_output(make_optional_bias(matmul, context, 2))};
+    auto weight = context.get_input(1);
+    if (weight.get_element_type() == element::f16 || weight.get_element_type() == element::bf16) {
+        // In case of patched linear it can have mixed fp16/bf16 and fp32 input type.
+        // In other cases these conversion is not required.
+        weight = context.mark_node(std::make_shared<ov::op::v1::ConvertLike>(weight, x));
```

Comment:
can we use `Convert` if the required type is known?

Reply:
`x` type may be still not known, so I don't think that is a good idea

Reply:
in TF FE we have helper that check if it is static or dynamic. Based on it, it uses either `Convert` or `ConvertLike`